### PR TITLE
Fix for Donut chart inner value bug

### PR DESCRIPTION
**Pull request by srmukher**
Work item https://uifabric.visualstudio.com/iss/_workitems/edit/9121/

Fix: Show the value inside donut chart same as the data selected/hovered. Show the value as per "valueInsideDonut" prop only when there is no active element (hovered/selected).

When "First" legend is selected:
image

When "Second" legend is selected:
image

When "Third" legend is selected:
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/44fc4425708d04018dbfa0875f453df324068fd7'>44fc4425708d04018dbfa0875f453df324068fd7</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: change/@fluentui-react-charting-75b91b91-57d5-4cc4-9e22-9e1009caa21f.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Fixing inner value change bug for donut",
```
Comment (AtishayMsft) [inline change/@fluentui-react-charting-75b91b91-57d5-4cc4-9e22-9e1009caa21f.json:3]: Make the comment more descriptive

File: packages/react-charting/src/components/DonutChart/DonutChart.base.tsx [hunk 2]
```diff
@@ -289,10 +289,10 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
 
   private _valueInsideDonut(valueInsideDonut: string | number | undefined, data: IChartDataPoint[]) {
     const highlightedLegend = this._getHighlightedLegend();
-    if (valueInsideDonut !== undefined && highlightedLegend !== '' && !this.state.showHover) {
+    if (valueInsideDonut !== undefined && (highlightedLegend !== '' || this.state.showHover)) {
       let legendValue = valueInsideDonut;
       data!.map((point: IChartDataPoint, index: number) => {
-        if (point.legend === highlightedLegend) {
+        if (point.legend === highlightedLegend || (this.state.showHover && point.legend === this.state.legend)) {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/DonutChart.base.tsx:295]: does this value represent the hovered element?

Comment (srmukher) [reply packages/react-charting/src/components/DonutChart/DonutChart.base.tsx:295]: yes, _hoverCallback handler is called on hovering over a particular pie of the donut, which sets the state.legend value. And _getHighlightedLegend() already sets the highlightedLegend based on whether a legend is selected or hovered (active legend).

File: change/@fluentui-react-charting-75b91b91-57d5-4cc4-9e22-9e1009caa21f.json (hunk 1 above)
Comment (srmukher) [reply change/@fluentui-react-charting-75b91b91-57d5-4cc4-9e22-9e1009caa21f.json:3]: done

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**